tests/ha/test_server_pod_failure.py

```diff
@@ -289,50 +247,28 @@ def test_degraded_reads_unsafe_server_pod_shutdown(self):
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Step 2: Performed READs and verified DI on the written data")
 
-        LOGGER.info("Step 3: Shutdown the server pod by deleting deployment (unsafe)")
-        LOGGER.info("Get server pod name to be deleted")
-        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
-        pod_name = random.sample(pod_list, 1)[0]
-        hostname = self.node_master_list[0].get_pod_hostname(pod_name=pod_name)
-
-        LOGGER.info("Deleting server pod %s", pod_name)
-        resp = self.node_master_list[0].delete_deployment(pod_name=pod_name)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_false(resp[0],
-                                  f"Failed to delete server pod {pod_name} by deleting deployment")
-        LOGGER.info(
-            "Step 3: Successfully shutdown/deleted server pod %s by deleting deployment", pod_name)
-        self.deployment_backup = resp[1]
-        self.deployment_name = resp[2]
+        LOGGER.info("Step 3: Shutdown random server pod safely by deleting deployment and "
+                    "verify cluster & remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX], down_method=const.RESTORE_DEPLOYMENT_K8S)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
+        pod_name = list(resp[1].keys())[0]
+        self.deployment_name = resp[1][pod_name]['deployment_name']
```

Comment:
Please check if we want to take deployment name list for restore, in case multiple pods are deleted.

Reply:
It seems this is for single pod only. Please ignore.

Reply:
> Please check if we want to take deployment name list for restore, in case multiple pods are deleted.
we have dict of dict which will contain data for all the pods, in multi, we have to loop through it